platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java

```diff
@@ -16,46 +16,97 @@
-     * @throws ParseException if there is any problem with creating an Address from the addressLine.
+     * @param addressBook the address book to serialize.
+     * @return the config.txt compatible text representation of the address book.
      */
     @NonNull
-    public static Address parseAddressConfigText(
-            @NonNull final String addressLine,
-            @NonNull final NodeId id,
-            @NonNull final Function<InetAddress, Boolean> isOwnHostDeterminer,
-            @NonNull final String memo)
-            throws ParseException {
-        Objects.requireNonNull(addressLine, "The addressLine must not be null.");
-        Objects.requireNonNull(id, "The id must not be null.");
-        Objects.requireNonNull(isOwnHostDeterminer, "The isOwnHostDeterminer must not be null.");
-        Objects.requireNonNull(memo, "The memo must not be null.");
-        final String[] parts = addressLine.trim().split(",");
-        if (parts.length < 8 || parts.length > 9) {
+    public static String addressBookConfigText(@NonNull final AddressBook addressBook) {
+        Objects.requireNonNull(addressBook, "The addressBook must not be null.");
+        final TextTable table = new TextTable().setBordersEnabled(false);
+        for (final Address address : addressBook) {
+            final String memo = address.getMemo();
+            final boolean hasMemo = !memo.trim().isEmpty();
+            final boolean hasInternalIpv4 = address.getAddressInternalIpv4() != null;
+            final boolean hasExternalIpv4 = address.getAddressExternalIpv4() != null;
+            table.addRow(
+                    "address,",
+                    address.getNodeId() + ",",
+                    address.getNickname() + ",",
+                    address.getSelfName() + ",",
+                    address.getWeight() + ",",
+                    (hasInternalIpv4 ? ipString(address.getAddressInternalIpv4()) : "") + ",",
+                    address.getPortInternalIpv4() + ",",
+                    (hasExternalIpv4 ? ipString(address.getAddressExternalIpv4()) : "") + ",",
+                    address.getPortExternalIpv4() + (hasMemo ? "," : ""),
+                    memo);
+        }
+        return table.render();
+    }
+
+    /**
+     * Parses an address book from text in the form described by config.txt.  Comments are ignored.
+     *
+     * @param addressBookText the config.txt compatible serialized address book to parse.
+     * @return a parsed AddressBook.
+     * @throws ParseException if any Address throws a ParseException when being parsed.
+     */
+    @NonNull
+    public static AddressBook parseAddressBookText(@NonNull final String addressBookText) throws ParseException {
+        Objects.requireNonNull(addressBookText, "The addressBookText must not be null.");
+        final AddressBook addressBook = new AddressBook();
+        for (final String addressLine : addressBookText.split("\\r?\\n")) {
```

Comment:
do we have the `addressBookText` defined somewhere. Otherwise I would like to see a sentence in the javadoc regarding the `addressBookText.split("\\r?\\n")` that shows a simple sample

Reply:
It's the format in config.txt.  It is described well there.  The split you're quoting splits on newlines between addresses.  The split handles both windows and the rest of the world formats for new lines.   I will add more comments to the javadoc.

Reply:
I have added a javadoc comment at the top of the `AddressBookUtils.java` that contains the parsing and serialization methods.